src/frontends/onnx/frontend/src/core/graph.cpp

```diff
@@ -382,14 +384,29 @@ void Graph::set_friendly_names(const Node& onnx_node, const OutputVector& ng_sub
         return;
     }
 
+    const auto common_node = detail::common_node_for_all_outputs(ng_subgraph_outputs);
+
     for (size_t i = 0; i < ng_subgraph_outputs.size(); ++i) {
         // Trailing optional outputs may not be specified in the ONNX model.
         // Other optional outputs should have name set to an empty string.
         if (i >= onnx_node.get_outputs_size()) {
             break;
         }
 
-        ng_subgraph_outputs[i].get_node()->set_friendly_name(onnx_node.output(i));
+        const auto& onnx_node_name = onnx_node.get_name();
+        if (onnx_node_name.empty()) {
+            // for multioutput nodes, their friendly name is always set to the last ONNX output's name
+            // this is because this setter is called in a loop and the last call is ultimate for a given node
+            ng_subgraph_outputs[i].get_node()->set_friendly_name(onnx_node.output(i));
+        } else {
+            if (common_node) {
+                ng_subgraph_outputs[i].get_node()->set_friendly_name(onnx_node.get_name());
```

Comment:
```suggestion
                ng_subgraph_outputs[i].get_node()->set_friendly_name(onnx_node_name);
```

Reply:
I'll fix that separately, the same with the suggestion below